lib/network_config.ml

```diff
@@ -114,7 +116,9 @@ let write_config config =
 let read_config () =
 	try
 		let config_json = Xapi_stdext_unix.Unixext.string_of_file config_file_path in
-		config_json |> Jsonrpc.of_string |> config_t_of_rpc
+		match config_json |> Jsonrpc.of_string |> Rpcmarshal.unmarshal typ_of_config_t with
```

Comment:
This is the part that we need to test very well: it needs to be able to read a JSON file with config that was written by the old version (pre-PPX) of the software, because that is what happens on upgrade.

Reply:
An issue here should make the update/upgrade tests in bvt and bst fail right?

Reply:
Most likely.

Reply:
Yes, I think it should be OK except the `interface_config_t.ipv4_routes` because it changes from `tuple list` to a `struct list`. 
I will test and add code to fix it.

Reply:
I have tested this case by following:
1. install XS of ring3 master branch
2. Configure `static_routes` for network by:
`xe network-param-set uuid=a09c23c3-9a46-e1aa-c7c8-6efb5b3256de other-config:static-routes="192.168.2.0/24/192.168.2.1"`
3. plug the PIF in the network
4. send `kill` signal to networkc process so it will persist the configuration to `networkd.db`
5. replace `xapi` and `xcp-networkd` by that of my private branch
6. restart toolstack

Expect result: We should get `Result.Error` when unmarshalling
Actual result: Got `Result.Ok`



Thd network.db file content is:
{"interface_config":{"xenbr0":{"ipv4_conf":"DHCP4","ipv6_conf":"Linklocal6",`"ipv4_routes":[["192.168.2.0",24,"192.168.2.1"]]`,"dns":[[],[]],"mtu":1500,"ethtool_settings":{},"ethtool_offload":{"gro":"on","lro":"off"},"persistent_i":true},"eth0":{"ipv4_conf":"None4","ipv6_conf":"None6","ipv4_routes":[],"dns":[[],[]],"mtu":1500,"ethtool_settings":{},"ethtool_offload":{"gro":"on","lro":"off"},"persistent_i":true},"eth1":{"ipv4_conf":"None4","ipv6_conf":"None6","ipv4_routes":[],"dns":[[],[]],"mtu":1500,"ethtool_settings":{},"ethtool_offload":{"gro":"on","lro":"off"},"persistent_i":true},"xenbr1":{"ipv4_conf":"None4","ipv6_conf":"Linklocal6","ipv4_routes":[],"dns":[[],[]],"mtu":1500,"ethtool_settings":{},"ethtool_offload":{"gro":"on","lro":"off"},"persistent_i":true}},"bridge_config":{"xenbr0":{"ports":{"eth0":{"interfaces":["eth0"],"bond_properties":{},"kind":"Basic"}},"bridge_mac":"c8:1f:66:ba:ff:2b","igmp_snooping":false,"other_config":{"static-routes":"192.168.2.0/24/192.168.2.1","memory-ratio-hvm":"0.25","memory-ratio-pv":"0.25","network-uuids":"a09c23c3-9a46-e1aa-c7c8-6efb5b3256de"},"persistent_b":true},"xenbr1":{"ports":{"eth1":{"interfaces":["eth1"],"bond_properties":{},"kind":"Basic"}},"bridge_mac":"c8:1f:66:ba:ff:2c","igmp_snooping":false,"other_config":{"memory-ratio-hvm":"0.25","memory-ratio-pv":"0.25","network-uuids":"92958b17-e115-0a17-acf8-12b2bbeea0c8"},"persistent_b":true}},"gateway_interface":"xenbr0","dns_interface":"xenbr0"}[

Reply:
Any thoughts ^ @robhoes @mseri ?

Reply:
I have updated this PR by implemented option 2.
https://github.com/xapi-project/xcp-networkd/pull/128/commits/090b340489cd42032cfbe230138330ce4b670e5a